src/DTO_validation/Processor/EmailAnnotationProcessor.java

```diff
@@ -0,0 +1,37 @@
+package DTO_validation.Processor;
+
+import DTO_validation.Annotation.Email;
+import DTO_validation.Annotation.Length;
+import DTO_validation.exceptions.AnnotationIncorrectUsageException;
+
+import java.lang.reflect.Field;
+import java.util.regex.Pattern;
+
+public class EmailAnnotationProcessor {
+
+    public static final String EMAIL_REGEX = "^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$";
+
+    public String[] emailAnnotationMeth(Object dto) throws NoSuchFieldException, IllegalAccessException {
+        Field[] declaredFields = dto.getClass().getDeclaredFields();
+        String[] errors = new String[0];
+        for (Field filed : declaredFields) {
+            if (filed.isAnnotationPresent(Email.class)) {
+                Email annotation = filed.getAnnotation(Email.class);
+                filed.setAccessible(true);
+                if (filed.get(dto) instanceof String) {
+                    String fieldValue = (String) filed.get(dto);
+                    int length = fieldValue.length();
+                    if (length != 0) {
+                        if (!fieldValue.matches(EMAIL_REGEX)) {
```

Comment:
Collapsible "if" statements should be merged.

Merging collapsible if statements increases the code’s readability.